src/collector.js

```diff
@@ -76,6 +77,19 @@ export default class Collector {
     return false;
   }
 
+  recordScannedFile(filename, scanner) {
+    // TODO: Add some code that verifies and normalizes `filename`
+    // to better avoid duplicates.
+    // See https://github.com/mozilla/addons-linter/issues/1310
+    if (filename in this.scannedFiles) {
```

Comment:
This will check the prototype chain too whereas `this.scannedFiles.hasOwnProperty(filename)` won't